scraper/ChunkScraper.ts

```diff
@@ -0,0 +1,102 @@
+import * as puppeteer from 'puppeteer'
+import { removeHtmlSpecials } from './helper'
+import { Chunk, CourseHead, CourseInfo, Career } from './interfaces'
+
+/**
+ * Gets the data from the title of the course (course code, name)
+ * @param { puppeteer.Page } page: page which displays the data to scrape
+ *
+ * @returns { Promise<CourseHead> }: Data about the title of the course: The course code and the course name
+ */
+const getCourseHeadData = async (page: puppeteer.Page): Promise<CourseHead> => {
+  // Get the course code and course name
+  const courseHead = await page.evaluate(() => {
+    const courseHeader = document.getElementsByClassName(
+      'classSearchMinorHeading'
+    )[0].innerHTML
+    const headerRegex = /(^[A-Z]{4}[0-9]{4})(.*)/
+    return headerRegex.exec(courseHeader)
+  })
+  // There must be at least 3 elements in courseHead
+  if (!courseHead && courseHead.length > 2) {
```

Comment:
Shouldn't this be a || ? Your first statement checks for an undefined/null `courseHead` object, which if indeed undefined/null will go into the next statement, But the next statement which grabs the `.length` property of the `courseHead` object won't go well because it's undefined/null?

And also if there must be at least 3 elements in the courseHead shouldn't the condition for FAILURE be `<= 2` NOT `> 2`?
```suggestion
  if (!courseHead || courseHead.length <= 2) {
```

Reply:
Yeah, I think I missed a bracket

Reply:
I assume you mean you missed a bracket by 

``` if (!(courseHead && courseHead.length > 2)) {```

In that case I think this is still better

``` if (!courseHead || courseHead.length <= 2) { ```

(See the other comment about conditions)